### DISCUSSION TITLE
联动时设置initValues 会出现闪烁

### DISCUSSION THREAD

**Discussion by wjwjwjw** [alibaba/formily#1430] [Q&A 2.x] [answered]
联动时设置initValues 会出现闪烁，城市会先显示value，闪烁后变为label值，是我代码哪里有问题么

![image](https://user-images.githubusercontent.com/33014303/118402963-d1731400-b69e-11eb-82c5-3fa499bca6de.png)

![image](https://user-images.githubusercontent.com/33014303/118402922-9bce2b00-b69e-11eb-9f43-c8762ed3bd4e.png)

![image](https://user-images.githubusercontent.com/33014303/118402905-8b1db500-b69e-11eb-9bb6-095506f28b4c.png)
![image](https://user-images.githubusercontent.com/33014303/118402937-b4d6dc00-b69e-11eb-82f1-a569d12e47db.png)

**Comment by liuweiGL** [comment]
给 codesandbox demo

**Reply by wjwjwjw** [reply]
https://codesandbox.io/s/damp-sun-wrj64

**Comment by janryWang** [comment, marked as answer]
你这个场景本来就应该是这样的，因为选择的时候，dataSource还没加载完，没办法映射的，肯定是先显示value

**Reply by wjwjwjw** [reply]
那这种有联动情况，筛选后怎么保持表单的值而不清空呢

**Reply by janryWang** [reply]
需求描述清楚点，什么是清空，什么是不清空

**Reply by wjwjwjw** [reply]
https://codesandbox.io/s/hopeful-swirles-6no67?file=/App.tsx
当我把form写在render里时，点击搜索表单值会清空，把form写到render外，我就获取不到异步的list了

**Comment by liuweiGL** [comment]
https://codesandbox.io/s/cranky-mendeleev-qbikr?file=/App.tsx

你的代码有问题吧，你在 render 里面 createForm 干嘛？

**Reply by wjwjwjw** [reply]
保存表单数据该怎么做呢？或者说 筛选后怎么保持表单的值而不清空呢

**Reply by liuweiGL** [reply]
不好意思，我也没明白你说的是什么意思。

**Reply by wjwjwjw** [reply]
https://codesandbox.io/s/hopeful-swirles-6no67?file=/App.tsx
当我把form写在render里时，点击搜索表单值会清空，把form写到render外，我就获取不到异步的list了

**Comment by liuweiGL** [comment]
https://codesandbox.io/s/dank-fast-6ioml?file=/App.tsx

你看看是不是期望的效果，写代码的思路可以调整一下，多思考一下哪些 UI + Data 是一个整体，然后放到组件里面去处理。

**Reply by liuweiGL** [reply]
一个消息不用多次回复，回复一次，大家都能收到消息的。

**Reply by wjwjwjw** [reply]
这种写法我有个新问题，每次表单change的时候，都会触发异步请求，如何避免频繁的请求呢

**Reply by liuweiGL** [reply]
不会吧，useEffect 使用空依赖的话只会触发一次啊。

**Comment by janryWang** [comment]
不要在一个问题里一直问其他问题，这种太不方便检索了
